server.go

```diff
@@ -661,6 +665,65 @@ func (s *Server) Publish(topic string, payload []byte, retain bool, qos byte) er
 	})
 }
 
+// Subscribe directly subscribes to one or more filters.
+func (s *Server) Subscribe(filter string, subscriptionId int, handler InlineSubFn) error {
+	if handler == nil {
+		return packets.ErrInlineSubscriptionHandlerInvalid
+	}
+
+	reasonCodes := make([]byte, 1)
+	if !IsValidFilter(filter, false) {
+		reasonCodes[0] = packets.ErrTopicFilterInvalid.Code
+	}
+	subscription := packets.Subscription{
+		Identifier: subscriptionId,
+		Filter:     filter,
+	}
+
+	pk := s.hooks.OnSubscribe(s.inlineClient, packets.Packet{ // subscribe like a normal client.
+		Origin:      s.inlineClient.ID,
+		FixedHeader: packets.FixedHeader{Type: packets.Subscribe},
+		Filters:     packets.Subscriptions{subscription},
+	})
+
+	inlineSubscription := InlineSubscription{
+		Subscription: subscription,
+		Handler:      handler,
+	}
+	s.Topics.InlineSubscribe(inlineSubscription)
```

Comment:
Not sure we should actually subscribe if `reasonCodes[0] == packets.ErrTopicFilterInvalid.Code` ?